app/presenters/details_presenter.rb

```diff
@@ -12,44 +12,75 @@ def initialize(content_item_details, change_history_presenter)
     def details
       @_details ||=
         begin
-          updated = content_item_details.each_with_object({}) do |(key, val), seed|
-            seed[key] = append_transformed_govspeak(val)
-          end
+          updated = recursively_transform_govspeak(content_item_details)
           updated[:change_history] = change_history unless change_history.blank?
           updated
         end
     end
 
   private
 
+    def recursively_transform_govspeak(obj)
```

Comment:
I was just trying some refactoring and you might be able to do something like: 

```ruby
    def govspeak?(value)
      value.all? { |hsh| hsh.is_a?(Hash) } &&
        value.one? { |hsh| hsh[:content_type] == "text/govspeak" } &&
        value.none? { |hsh| hsh[:content_type] == "text/html" }
    end

    def process(value)
      return unless value
      wrapped_array = Array.wrap(value)
      return render_govspeak(value) if govspeak?(wrapped_array)
      return value if value.is_a?(String)
      return value if value.respond_to?(:has_key?) && value.has_key?(:content)
      value.map {|o| recursively_transform_govspeak(o) }
    end

    def recursively_transform_govspeak(obj)
      Array(obj).each_with_object({}) do |(key, value), memo|
        memo[key] = process(value)
      end
    end
```

it feels little neater then the multi elsif, (but only a little)